cdap-explore/src/main/java/co/cask/cdap/explore/service/ExploreServiceUtils.java

```diff
@@ -205,7 +208,13 @@ public static HiveSupport checkHiveSupport(@Nullable ClassLoader hiveClassLoader
     }
 
     ClassLoader classLoader = ExploreUtils.getExploreClassloader();
-    return traceExploreDependencies(classLoader, tmpDir);
+    Set<File> additionalJars = new HashSet<>();
+    if (isSparkAvailable()) {
```

Comment:
The engine can be selected dynamically. So, I'm shipping the jar if it is available.